kong/clustering/wrpc_control_plane.lua

```diff
@@ -235,16 +235,25 @@ function _M:handle_cp_websocket()
   end
 
   do
-    local _, err
-    _, err, client.sync_status = self:check_version_compatibility(dp_version, client.dp_plugins_map, log_suffix)
+    local ok, err = client.basic_info_semaphore:wait(5)
+    if not ok then
+      err = "waiting for basic info call: " .. (err or "--")
+    end
+    if not client.basic_info then
+      err = "invalid basic_info data"
+    end
+
     if err then
       ngx_log(ngx_ERR, _log_prefix, err, log_suffix)
       wb:send_close()
-      client:update_sync_status()
       return ngx_exit(ngx_CLOSE)
     end
   end
 
+  -- after basic_info report we consider DP connected
+  -- initial sync
+  self.clients[w_peer.conn] = client
```

Comment:
We move the line registering clients to list here, so the DP will not be pushed config before it finishes its first meta report.
Fix No.2